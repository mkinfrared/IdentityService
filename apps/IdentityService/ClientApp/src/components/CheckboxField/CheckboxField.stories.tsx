import { Meta, Story } from "@storybook/react";

import { CheckboxField } from "./CheckboxField";
import css from "./CheckboxField.module.scss";
import { CheckboxFieldProps } from "./CheckboxField.type";

export default {
  title: "COMPONENTS/CheckboxField",
  component: CheckboxField,
  parameters: {
    componentSubtitle: "Subtitle goes here",
  },
} as Meta;

const Template: Story<CheckboxFieldProps> = (args) => (
  <div className={css.Story}>
    <CheckboxField {...args} />
  </div>
);

const Default = Template.bind({});

Default.args = {};

Default.parameters = {
  docs: {
    storyDescription: "Story description",
  },
};

export { Default };
