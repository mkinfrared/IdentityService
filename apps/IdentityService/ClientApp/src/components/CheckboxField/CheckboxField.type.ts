import { CheckboxProps } from "@identity-service/ui";

export type CheckboxFieldProps = Omit<CheckboxProps, "inputRef"> & {
  /**
   * a string that will be applied as a css class to parent element
   */
  className?: string;
};
