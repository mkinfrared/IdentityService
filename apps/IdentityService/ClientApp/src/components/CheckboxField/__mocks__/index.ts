export { default } from "./CheckboxField.mock";
