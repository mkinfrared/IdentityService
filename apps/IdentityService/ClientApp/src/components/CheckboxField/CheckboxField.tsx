import { Checkbox } from "@identity-service/ui";
import { Ref, forwardRef, memo } from "react";

import { CheckboxFieldProps } from "./CheckboxField.type";

const CheckboxField = forwardRef(
  (props: CheckboxFieldProps, ref: Ref<HTMLInputElement>) => (
    <Checkbox {...props} inputRef={ref} />
  ),
);

CheckboxField.displayName = "Checkbox";

export { CheckboxField };

export default memo(CheckboxField);
