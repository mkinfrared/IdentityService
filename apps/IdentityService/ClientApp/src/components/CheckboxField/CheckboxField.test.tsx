import { render } from "@testing-library/react";

import { CheckboxField } from "./CheckboxField";

describe("<CheckboxField />", () => {
  const Component = <CheckboxField />;

  it("should be defined", () => {
    expect(CheckboxField).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("Checkbox");

    expect(element).toBeDefined();
  });
});
