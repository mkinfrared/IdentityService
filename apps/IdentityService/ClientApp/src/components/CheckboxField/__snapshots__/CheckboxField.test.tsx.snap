// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckboxField /> should match the snapshot 1`] = `
<div>
  <label
    class="Checkbox-module_Checkbox__ChGlN"
    data-testid="Checkbox"
  >
    <input
      class="Checkbox-module_input__-039s"
      type="checkbox"
      value=""
    />
    <div
      class="Checkbox-module_checkmark__pjJT2"
    >
      <svg
        class="Checkbox-module_tick__-Yp8y"
        viewBox="0 0 1280 1186"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1253 5.4c-3 1.8-14.5 8.6-25.5 15.1C1114 87.6 990.7 180.8 869 291.4c-64.7 58.9-140.3 134.2-203 202.1-108.5 117.6-218 252.3-311.3 382.9l-4.8 6.7-3.2-1.9c-1.8-1.1-75.8-43-164.6-93.2-130.9-73.9-161.6-90.9-162.6-89.8-6.2 6.8-15.6 19.1-15.3 20 .2.6 90.8 106.1 201.4 234.4l200.9 233.3h7.5l7.6.1 15.3-30.8c174.3-350.6 357.9-626.1 584-876.7 66-73.1 162.4-170.1 238.1-239.4 8.5-7.8 15.4-14.8 15.4-15.6-.2-1.5-14.5-21.4-15.4-21.4-.3 0-3 1.4-6 3.3z"
        />
      </svg>
    </div>
  </label>
</div>
`;
